### DISCUSSION TITLE
UseField hook : add a "resetDirty" callback

### DISCUSSION THREAD

**Discussion by dev-banana** [mantinedev/mantine#6346] [Feature requests]
Hello !

For the "use-field" hook, I was wondering if it was possible to add a function to reset the dirty state to a specific value, as is possible for the "useForm" hook ?

```typescript
const field = useField({
  initialValue: initialValue,
});

const onCustomReset = (newInitialValue) => {
  field.resetDirty(newInitialValue)
}
```

In terms of implementation, I was thinking something like this :
```typescript
function useField({
  // ...
  initialValue,
  // ...
}) {
  const [localInitialValue, setLocalInitialValue] = react.useState(initialValue); // Use a state to manage the initialValue allows us to reset it
  const [valueState, setValueState] = react.useState(initialValue);
  // ...
  const reset = react.useCallback(() => {
    setValue(localInitialValue); // Use the local state instead
    // ...
  }, [localInitialValue]);
  // ...
  const isDirty = react.useCallback(() => valueRef.current !== localInitialValue, [localInitialValue]); // Use the local state instead
  // ...
  const resetTouched = react.useCallback(() => setTouched(false), []);
  const resetDirty = react.useCallback((newInitialValue) => setLocalInitialValue(newInitialValue), []); // reset the initialValue with a custom value
  return {
    // ...
    resetDirty
  };
}
```

Thank you !

**Comment by dev-banana** [comment]
I try an implementation here :
[https://github.com/dev-banana/mantine/commit/90aa3e552c1700785b136927174fdb898a1ea3d1](https://github.com/dev-banana/mantine/commit/90aa3e552c1700785b136927174fdb898a1ea3d1)

**Comment by rtivital** [comment]
There is no need for `resetDirty` handler. `initialValue` works differently with useField hook compared to useForm – it accepts dynamic values and the field dirty status is always up-to-date with the updated initial value.
